### ISSUE TITLE
Multi tag selection

### ISSUE THREAD

**Issue by Anzan** [fabiospampinato/notable#398]
### Feature description

Select multiple tags in the left menu in order to display only notes that contain them all

### Feature motivation

Especially since Notebooks are actually just tags, I sometimes want to restrict displaying Notes that are in a specific Notebook and containing a specific Tag. In other situations, being able to choose more than one Tag is a powerful way to be very specific, especially if you have a lot of notes, like a long list of various code snippets. For example: display notes that are in Notebook Snippets, with Tags PHP and SQL. You can do this with Evernote using the ctrl key.

**Comment by fabiospampinato** [comment]
I think this will be solved once we have advanced search operators support #293.

> You can do this with Evernote using the ctrl key.

Can you post a screenshot or something? ctrl+click opens up the right-click menu for me under macOS, and shift/alt/cmd-click all reset the search.

**Comment by Anzan** [comment]
Sorry, I should have mentioned I'm using Windows 10, I don't have a Mac. After Googling a bit, there doesn't seem to be an easy way to do this on Mac, not sure why. Here's a screenshot on Windows for finding a Note selecting both PHP and Regex Tags on the left, it's an easy and seamless operation.

![multi-tags](https://user-images.githubusercontent.com/4517616/52145438-f27adc80-262e-11e9-947e-9e7c822208ac.png)
.

**Comment by fabiospampinato** [comment]
Thanks for the screenshot. It's weird that there doesn't seem to be a way to do this under macOS.

**Comment by roccowang** [comment]
> Thanks for the screenshot. It's weird that there doesn't seem to be a way to do this under macOS.

under macos this can be done when you are in 'tags' page.
![image](https://user-images.githubusercontent.com/9870918/52626030-48803780-2eed-11e9-9629-6a4cd0b32484.png)
hold cmd then select some tags, and you'll get a 'show xx note' button in the upper right corner

**Comment by adamshand** [comment]
I'd like this as well.  I use tags to indicate the status of things I'm writing (eg. idea, doing, done, posted, abandoned).  So being able to select Writing/Blog and Status/Doing would be a really lovely way to see what I'm currently working on. Or Writing/Blog and Status/Done would give me a list of things waiting to be posted to my blog.

**Comment by fabiospampinato** [comment]
I've recently implemented this in my development branch, but I've just now realized that tags are joined by an OR, while instead they should probably be joined by an AND 🤔 Ideally this should actually be customizable, but I'm not sure how that can be implemented UX-wise.

**Comment by adamshand** [comment]
Good point.  If it can only be one I vote for AND because it's easy enough to click on two different tags to see if something I'm looking for is there.  But there's currently no easy way to see which items have a set of tags.

If you want to allow for either behaviour, my suggestion would be to control it with the meta key used for selection, Eg. 

* If I hold down Ctrl to multiple select then I get AND behaviour.
* If I hold down Shift-Ctrl to multiple select then I get OR behaviour.

If you wanted to take it one step further you could have a preference which allows me to choose which way around I want the behaviour (ie. Shift required for AND or for OR).

**Comment by fabiospampinato** [comment]
That could work, but how do we display if tags are being joined by an AND or an OR in the interface? 🤔

**Comment by adamshand** [comment]
Personally I think you probably don't need to.  If the behaviour was just one way then people would figure out what it was doing and get used to it.

If you want to indicate visually somehow then my best idea is to use a different highlight colour for the selected tags (azure for AND, orange for OR ... or whatever :-).

**Comment by skostojohn** [comment]
Just adding that I think this feature is important as well.  I want to see notes that are (a) in a given notebook and (b) with a given tag.  I don't think that I can do this in Notable today.

**Comment by fabiospampinato** [comment]
@skostojohn Yeah you can't do that in v1.8 unfortunately. Consider this feature implement for v1.9, there's only the very minor detail mentioned above to adjust. If things go smoothly actually in v1.9 one should be able to search notes in way more complex ways too.
